x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/Phase.java

```diff
@@ -183,4 +176,19 @@ public String toString() {
         return Strings.toString(this, true, true);
     }
 
+    @UpdateForV10(owner = UpdateForV10.Owner.DATA_MANAGEMENT)
+    public boolean maybeAddDeprecationWarningForFreezeAction(String policyName) {
+        if (getActions().containsKey(FreezeAction.NAME)) {
+            deprecationLogger.warn(
+                DeprecationCategory.OTHER,
+                "ilm_freeze_action_deprecation",
+                "The freeze action in ILM is deprecated and will be removed in a future version. Please remove the freeze action from the '"
```

Comment:
We should mention that it is a noop also, so a user knows removing it doesn't affect anything